### PR TITLE
75 howto100m preprocessing

### PR DESCRIPTION
**Pull request by cwmeijer**
This will add howto100m preprocessing to platalea.

### CODE REVIEW

File: platalea/utils/preprocessing.py [hunk 1]
```diff
@@ -17,13 +21,38 @@
...
+    video_path = pathlib.Path(dataset_path) / video_subdir
+    if not audio_path.exists():
+        extract_audio_from_videos(video_path, audio_path)
+
+    extract_howto100m_audio_features(dataset_path, audio_subdir, _audio_feat_config)
+
+
+def extract_howto100m_audio_features(dataset_path, audio_subdir, feat_config):
+    audio_dir_path = dataset_path / audio_subdir
+    file_names = os.listdir(audio_dir_path)
```
Comment (bhigy) [inline platalea/utils/preprocessing.py:69]: as `audio_dir_path` is a `pathlib.Path`, you can directly do `audio_dir_path.iterdir()`

File: .github/workflows/pythonapp.yml [hunk 2]
```diff
@@ -15,6 +15,8 @@ jobs:
         python-version: 3.8
     - name: Install tox and coverage
       run: pip install tox coverage
+    - name: Install libsndfile1-dev
```
Comment (bhigy) [inline .github/workflows/pythonapp.yml:18]: is that related to moviepy?

File: platalea/utils/preprocessing.py [hunk 3]
```diff
@@ -17,13 +21,38 @@
...
+    paths = [audio_dir_path / fn for fn in file_names]
+    features = audio_features(paths, feat_config)
+    torch.save(dict(features=features, filenames=file_names), dataset_path / 'mfcc_features.pt')
```
Comment (bhigy) [inline platalea/utils/preprocessing.py]: aren't we supposed to save that as a numpy memmap?

Comment (cwmeijer) [reply platalea/utils/preprocessing.py]: good point of course

File: .github/workflows/pythonapp.yml (hunk 2 above)
Comment (cwmeijer) [reply .github/workflows/pythonapp.yml:18]: yeah, it's in the documentation of moviepy. It won't install otherwise.

File: platalea/utils/preprocessing.py (hunk 1 above)
Comment (cwmeijer) [reply platalea/utils/preprocessing.py:69]: Thanks, I'm not used to using pathlib.Path. This is a nice helper function.

**Comment by egpbos** [comment]
Is this ready to merge?

File: requirements.txt [hunk 4]
```diff
@@ -1,3 +1,4 @@
+moviepy
```
Comment (egpbos) [inline requirements.txt:1]: This should probably be in setup.py, so here it can be removed. `sndfile` should also be in setup.py (not sure if it is).

We decided at some point that we would use requirements.txt only to pin specific dependency versions for reproducibility of specific experiments as published in some paper or at a conference.

File: tox.ini [hunk 5]
```diff
@@ -12,6 +12,8 @@ deps =
     coverage
     scikit-learn
     python-Levenshtein
+    moviepy
+    sndfile
```
Comment (egpbos) [inline tox.ini]: These two (moviepy and sndfile) can be removed here. If they are in setup.py, they will automatically be installed via pip in the tox run.

**Review by egpbos — CHANGES_REQUESTED** [review]
Some dependency reshuffling, otherwise LGTM.

File: tox.ini (hunk 5 above)
Comment (egpbos) [reply tox.ini]: Actually, I think scikit-learn and python-Levenshtein should also be removed here, they should also be installed through pip via setup.py. Could you try that?

**Review by bhigy — APPROVED** [review]

**Comment by cwmeijer** [comment]
Thanks for reviewing. I'll take it into account. I put this thing back into draft mode though. I'm at the same time working on the data processing (as opposed to this PR about PREprocessing). Many problems that I encounter in the processing, can best be solved by changing the preprocessing. When I think it all makes sense and is functional, I'll ask you (or 1 of you) to have another look at the final changes.

File: requirements.txt (hunk 4 above)
Comment (cwmeijer) [reply requirements.txt:1]: I've moved those dependencies around between setup and even tox because of your suggestion below but it kept giving me errors. I'm sure something can be streamlined here, but I could not make it work the way you suggested (which would have been the most clean way, I agree)

**Comment by cwmeijer** [comment]
I fixed the issues raised by Patrick some time ago. I also changed many other things. For instance, there's an index file being written that contains some metadata of start/end locations in the memmap and paths to video features. I think it's ready for merge now, so I requested another review.
